src/cluster_asm.c

```diff
@@ -2902,30 +2870,61 @@ void asmActiveTrimCycle(int type) {
         timelimit = server.asm_trim_fast_cycle_duration; /* in microseconds. */
     }
 
-    do {
-        cursor = kvstoreScan(server.db[0].keys, cursor, -1,
-                             asmActiveTrimScanCallback,
-                             asmActiveTrimShouldSkipDict,
-                             &server.db[0]);
-        /* Once in 16 scan iterations, 32 deletions
-         * (if we have a lot of keys in one hash bucket or rehashing),
-         * check if we reached the time limit. */
-        if (cursor && (++iterations > 16 || asmManager->active_trim_keys_deleted - prev_trimmed > 32)) {
-            if ((ustime() - start) > timelimit)
+    /* Initialize the slots array if this is the first call. */
+    if (current_slot == -1) {
+        current_slot = 0;
+        memset(slots, 0, sizeof(slots));
+
+        slotRangeArray *sra = listNodeValue(listFirst(asmManager->active_trim_jobs));
```

Comment:
how about just accessing `sra`, like doing somethings in `slotRangesSnapshotSaveRio`, not sure if i miss somethings, if so, please revert, maybe it is easy to merge to fork.

Reply:
see the last commit, I guess this is cleaner.

Reply:
done